In [80]:
# following Charlene's tutorial 
# the goal is to find genes essential to HepG2 (cancer cell line)

import pandas as pd
from pybedtools import BedTool
dep_score = pd.read_csv('D2_DRIVE_gene_dep_scores.csv')
dep_score

Unnamed: 0  127399_SOFT_TISSUE  1321N1_CENTRAL_NERVOUS_SYSTEM  \
0              CDH2 (1000)           -0.089352                      -0.002663   
1             AKT3 (10000)           -0.131082                       0.065231   
2     POU5F1P5 (100009667)           -0.212644                      -0.547079   
3             MED6 (10001)           -0.225475                      -0.477512   
4            NR2E3 (10002)           -0.096706                      -0.123476   
...                    ...                 ...                            ...   
7970         PPP4R1 (9989)           -0.036766                       0.172239   
7971            CDH1 (999)           -0.174443                       0.053323   
7972          PTBP3 (9991)           -0.036799                       0.001485   
7973          KCNE2 (9992)           -0.020674                       0.003644   
7974       CASP8AP2 (9994)           -0.762784                      -0.224090   

      2313287_STOMACH  697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  769P_KIDNEY  \
0           -0.137925                                0.173165    -0.134545   
1            0.087221                               -0.411486     0.177895   
2           -0.159532                               -0.277817     0.179144   
3           -0.314678                               -0.192239    -0.359350   
4           -0.138421                               -0.873816    -0.043431   
...               ...                                     ...          ...   
7970         0.138488                               -0.212066     0.181223   
7971        -0.362161                               -0.123080     0.007448   
7972        -0.361670                                0.187109    -0.291701   
7973         0.002908                               -0.080498     0.154795   
7974        -0.208999                               -0.428950     0.018032   

      786O_KIDNEY  8305C_THYROID  8505C_THYROID  921_UVEA  ...  \
0       -0.148781      -0.203150      -0.037625 -0.175778  ...   
1        0.324306       0.040381       0.022989 -0.360923  ...   
2        0.084569      -0.179724       0.000420 -0.091683  ...   
3       -0.705736      -0.346217      -0.355164 -0.510637  ...   
4       -0.200512      -0.181933      -0.192233  0.064916  ...   
...           ...            ...            ...       ...  ...   
7970     0.202738      -0.043067       0.082625  0.179550  ...   
7971     0.073178      -0.081611       0.029975  0.147575  ...   
7972    -0.440892      -0.082666      -0.346774 -0.544378  ...   
7973     0.059115       0.178424       0.057289  0.063665  ...   
7974     0.016122      -0.898269      -0.550961 -0.580094  ...   

      UMUC3_URINARY_TRACT  VMRCRCW_KIDNEY  WM115_SKIN  WM1799_SKIN  \
0               -0.066021        0.107229   -0.227427    -0.032107   
1               -0.035967       -0.031698    0.221841     0.207119   
2                0.106033        0.060357    0.137065     0.072449   
3               -0.249799       -0.773315   -0.403044    -0.241388   
4               -0.205856        0.024931    0.276181     0.093556   
...                   ...             ...         ...          ...   
7970             0.012154        0.003735   -0.222169     0.046680   
7971            -0.001828        0.018768    0.064089     0.112538   
7972            -0.279045       -0.190559   -1.071656    -0.196719   
7973             0.152772        0.045208    0.131550     0.261826   
7974            -0.247897       -0.485356   -0.168850    -0.220189   

      WM2664_SKIN  WM793_SKIN  WM88_SKIN  \
0       -0.218693   -0.013857        NaN   
1       -0.175644   -0.063389        NaN   
2        0.274616    0.032009        NaN   
3       -0.154820   -0.564204  -0.267406   
4       -0.119268    0.046175   0.406302   
...           ...         ...        ...   
7970     0.031706    0.085820        NaN   
7971    -0.010692    0.188347        NaN   
7972    -0.699364   -0.610545  -1.166492   
7973     0.106279    0.006334  -

In [81]:
# split the names out
dep_score['Symbol'] = dep_score['Unnamed: 0'].str.split(' ', expand = True)[0]

In [82]:
# drop the unneeded column and make the index Symbol (labels)
dep_score.drop('Unnamed: 0', inplace = True, axis = 1)
dep_score.set_index('Symbol', inplace = True)
dep_score

127399_SOFT_TISSUE  1321N1_CENTRAL_NERVOUS_SYSTEM  2313287_STOMACH  \
Symbol                                                                         
CDH2               -0.089352                      -0.002663        -0.137925   
AKT3               -0.131082                       0.065231         0.087221   
POU5F1P5           -0.212644                      -0.547079        -0.159532   
MED6               -0.225475                      -0.477512        -0.314678   
NR2E3              -0.096706                      -0.123476        -0.138421   
...                      ...                            ...              ...   
PPP4R1             -0.036766                       0.172239         0.138488   
CDH1               -0.174443                       0.053323        -0.362161   
PTBP3              -0.036799                       0.001485        -0.361670   
KCNE2              -0.020674                       0.003644         0.002908   
CASP8AP2           -0.762784                      -0.224090        -0.208999   

          697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  769P_KIDNEY  786O_KIDNEY  \
Symbol                                                                       
CDH2                                    0.173165    -0.134545    -0.148781   
AKT3                                   -0.411486     0.177895     0.324306   
POU5F1P5                               -0.277817     0.179144     0.084569   
MED6                                   -0.192239    -0.359350    -0.705736   
NR2E3                                  -0.873816    -0.043431    -0.200512   
...                                          ...          ...          ...   
PPP4R1                                 -0.212066     0.181223     0.202738   
CDH1                                   -0.123080     0.007448     0.073178   
PTBP3                                   0.187109    -0.291701    -0.440892   
KCNE2                                  -0.080498     0.154795     0.059115   
CASP8AP2                               -0.428950     0.018032     0.016122   

          8305C_THYROID  8505C_THYROID  921_UVEA  A101D_SKIN  ...  \
Symbol                                                        ...   
CDH2          -0.203150      -0.037625 -0.175778    0.139467  ...   
AKT3           0.040381       0.022989 -0.360923   -0.094837  ...   
POU5F1P5      -0.179724       0.000420 -0.091683    0.439111  ...   
MED6          -0.346217      -0.355164 -0.510637   -0.244747  ...   
NR2E3         -0.181933      -0.192233  0.064916    0.163630  ...   
...                 ...            ...       ...         ...  ...   
PPP4R1        -0.043067       0.082625  0.179550   -0.044767  ...   
CDH1          -0.081611       0.029975  0.147575    0.091671  ...   
PTBP3         -0.082666      -0.346774 -0.544378   -0.417071  ...   
KCNE2          0.178424       0.057289  0.063665    0.045011  ...   
CASP8AP2      -0.898269      -0.550961 -0.580094   -0.362667  ...   

          UMUC3_URINARY_TRACT  VMRCRCW_KIDNEY  WM115_SKIN  WM1799_SKIN  \
Symbol                                                                   
CDH2                -0.066021        0.107229   -0.227427    -0.032107   
AKT3                -0.035967       -0.031698    0.221841     0.207119   
POU5F1P5             0.106033        0.060357    0.137065     0.072449   
MED6                -0.249799       -0.773315   -0.403044    -0.241388   
NR2E3               -0.205856        0.024931    0.276181     0.093556   
...                       ...             ...         ...          ...   
PPP4R1               0.012154        0.003735   -0.222169     0.046680   
CDH1                -0.001828        0.018768    0.064089     0.112538   
PTBP3               -0.279045       -0.190559   -1.071656    -0.196719   
KCNE2                0.152772        0.045208    0.131550     0.261826   
CASP8AP2            -0.247897       -0.485356   -0.168850    -0.220189   

          WM2664_SKIN  WM793_SKIN  WM88_SKIN  \
Symbol                                       

In [83]:
# get hepg2 targets by making a hepg2 df
hepg2 = dep_score[['HEPG2_LIVER']]
hepg2

HEPG2_LIVER
Symbol               
CDH2         0.057906
AKT3        -0.269006
POU5F1P5    -0.040908
MED6        -0.346157
NR2E3       -0.200507
...               ...
PPP4R1      -0.041508
CDH1        -0.207714
PTBP3       -0.188387
KCNE2        0.041099
CASP8AP2     0.013360

[7975 rows x 1 columns]

In [84]:
targets = hepg2.loc[hepg2['HEPG2_LIVER'] < -1].index
type(targets)
# index is a sequence used for sequencing, immutable like string

pandas.core.indexes.base.Index

In [85]:
targets

Index(['DCAF13P3', 'SMC4', 'ATP6V1G2-DDX39B', 'ALYREF', 'SF3B4',
       'U2AF1L5&U2AF1', 'SF3A1', 'RACK1', 'PPIE', 'TACC3',
       ...
       'CCNK', 'BUD31', 'RPL14', 'PRPF4', 'PRPF3', 'COPB2', 'ZRANB2', 'CKAP5',
       'CDK11B', 'RBX1'],
      dtype='object', name='Symbol', length=139)

In [86]:
# splitting out the & from targets 
essential_gene = []
for t in targets:
    if '&' in t:
        essential_gene.extend(t.split('&'))
        print(t)
    else:
        essential_gene.append(t)

U2AF1L5&U2AF1
GPR89B&GPR89A


In [87]:
essential_gene
# these are the transcripts bound
# in ENCODE, we have RBPs 

['DCAF13P3',
 'SMC4',
 'ATP6V1G2-DDX39B',
 'ALYREF',
 'SF3B4',
 'U2AF1L5',
 'U2AF1',
 'SF3A1',
 'RACK1',
 'PPIE',
 'TACC3',
 'NXF1',
 'CCT7',
 'CCT2',
 'SMC2',
 'TBL3',
 'USP39',
 'RUVBL2',
 'WDR3',
 'AFG3L2',
 'SF3B2',
 'RPL35',
 'U2AF2',
 'AP2M1',
 'CLNS1A',
 'CLTC',
 'NEDD1',
 'COPA',
 'CTNNB1',
 'CYP21A1P',
 'DDB1',
 'DHX8',
 'DHX15',
 'DYNC1I2',
 'DUSP9',
 'EEF2',
 'COPZ1',
 'PUF60',
 'SNRNP200',
 'BOP1',
 'SF3B1',
 'NR5A1',
 'XRCC6',
 'GAK',
 'GEMIN5',
 'PRPF31',
 'HTATSF1',
 'PRPF19',
 'NOP53',
 'HNRNPC',
 'HNRNPU',
 'HSPE1',
 'IK',
 'KIF11',
 'SBK1',
 'MCM6',
 'CRSP8P',
 'NACA',
 'NRAS',
 'ACO2',
 'PABPC3',
 'DDX47',
 'CDC40',
 'DDX41',
 'GPR89B',
 'GPR89A',
 'SF3B6',
 'ATP6V0C',
 'POLA1',
 'POLR2A',
 'POLR2B',
 'POLR2J',
 'INO80',
 'SMU1',
 'PPP6C',
 'RBM22',
 'PSMA3',
 'PSMA4',
 'PSMC1',
 'PSMC2',
 'PSMC3',
 'PSMC5',
 'PSMD2',
 'NAT14',
 'RBM25',
 'RAN',
 'ABCE1',
 'RPA1',
 'RPA2',
 'RPL7',
 'RPS16',
 'RPS18',
 'RPS21',
 'RPS27A',
 'RRM1',
 'S100A1',
 'SNRNP70',
 'SNRPB',
 'S

## Finding the coordinate for essential genes

In [88]:
# loading transcript annotations
transcript_coords = BedTool('/home/hsher/gencode_coords/gencode.v33.transcript.gff3')


In [89]:
len(transcript_coords)

60134

In [90]:
print(transcript_coords[0])
# each entry is a transcript
# the gene symbol occurs in the transcript_name attribute

chr1	HAVANA	transcript	11869	14409	.	+	.	ID=ENST00000456328.2;Parent=ENSG00000223972.5;gene_id=ENSG00000223972.5;transcript_id=ENST00000456328.2;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2;havana_transcript=OTTHUMT00000362751.1



In [91]:
transcript_coords[0].attrs['transcript_name']

'DDX11L1-202'

In [92]:
# look into bedtools filter and lambda
# filtering based on transcript name
essential_gene_coords = transcript_coords.filter(lambda t: t.attrs['transcript_name'] in essential_gene).saveas()

In [93]:
# we created new BedTool to hold the coords
type(essential_gene_coords)

pybedtools.bedtool.BedTool

In [94]:
# the genes are named differently so need to convert to ensembl transcript ID
import mygene
mg = mygene.MyGeneInfo()
converted = mg.querymany(essential_gene, scopes='symbol',
    fields='ensembl.gene,entrezgene', as_dataframe = True, species = 9606)

querying 1-141...done.
Finished.
3 input query terms found dup hits:
	[('DCAF13P3', 2), ('ATP6V1G2-DDX39B', 2), ('CYP21A1P', 5)]
2 input query terms found no hit:
	['VARS', 'HIST1H2BO']
Pass "returnall=True" to return complete lists of duplicate or missing query terms.


In [95]:
# getting stuck at pandas module must be installed for as_dataframe
# the solution was to restart the kernel after installing pandas

In [96]:
converted.head()

_id      _score     ensembl.gene entrezgene  \
query                                                                      
DCAF13P3         ENSG00000259378   38.585747  ENSG00000259378        NaN   
DCAF13P3               100132724   38.585747              NaN  100132724   
SMC4                       10051   86.033905  ENSG00000113810      10051   
ATP6V1G2-DDX39B  ENSG00000254870  115.094610  ENSG00000254870        NaN   
ATP6V1G2-DDX39B        100532737  112.499650              NaN  100532737   

                ensembl notfound  
query                             
DCAF13P3            NaN      NaN  
DCAF13P3            NaN      NaN  
SMC4                NaN      NaN  
ATP6V1G2-DDX39B     NaN      NaN  
ATP6V1G2-DDX39B     NaN      NaN

In [97]:
# turning the essential gene ids to a list
essential_gene_ids = converted['ensembl.gene'].dropna().tolist()

In [98]:
# these are both BedTools
# FILTER function is useful here
essential_gene_coords = transcript_coords.filter(lambda x: x.attrs['Parent'].split('.')[0] in essential_gene_ids).saveas()

In [99]:
print(len(essential_gene_ids))
print(type(essential_gene_ids))


134
<class 'list'>


- found that the length of my essential gene IDs list is different than the reference (mine was 134, reference had 137)
- my mygene query also returned different results, but not sure why

In [100]:
len(essential_gene_coords)
# this value is the same as the reference notebook!

128

## Looking at IDR peaks of essential RBPs

In [101]:
# using ENCODE data
encode4 = pd.read_csv('/home/hsher/projects/ClipNet/archishma/ENCODE4_1214.csv')
encode3 = pd.read_csv('/home/hsher/projects/RBP_annot/ENCODE_FINAL_ANNOTATIONS.uidsonly.txt.manifesthg38.txt', sep = '\t', header= 0)

In [102]:
encode4.head()

Unnamed: 0   uid   Batch     RBP               prefix  \
0           0  4001  batch2    NONO  encode4_batch2.4001   
1           1  4002  batch2    FXR2  encode4_batch2.4002   
2           2  4004  batch2     DEK  encode4_batch2.4004   
3           3  4007  batch2  EIF2B3  encode4_batch2.4007   
4           4  4008  batch2  RBFOX3  encode4_batch2.4008   

                                               bam_0  \
0  /projects/ps-yeolab5/encore/processing/encore_...   
1  /projects/ps-yeolab5/encore/processing/encore_...   
2  /projects/ps-yeolab5/encore/processing/encore_...   
3  /projects/ps-yeolab5/encore/processing/encore_...   
4  /projects/ps-yeolab5/encore/processing/encore_...   

                                               bam_1  \
0  /projects/ps-yeolab5/encore/processing/encore_...   
1  /projects/ps-yeolab5/encore/processing/encore_...   
2  /projects/ps-yeolab5/encore/processing/encore_...   
3  /projects/ps-yeolab5/encore/processing/encore_...   
4  /projects/ps-yeolab5/encore/processing/encore_...   

                                         bam_control  \
0  /projects/ps-yeolab5/encore/processing/encore_...   
1  /projects/ps-yeolab5/encore/processing/encore_...   
2  /projects/ps-yeolab5/encore/processing/encore_...   
3  /projects/ps-yeolab5/encore/processing/encore_...   
4  /projects/ps-yeolab5/encore/processing/encore_...   

                                              plus_0  \
0  /projects/ps-yeolab5/encore/processing/encore_...   
1  /projects/ps-yeolab5/encore/processing/encore_...   
2  /projects/ps-yeolab5/encore/processing/encore_...   
3  /projects/ps-yeolab5/encore/processing/encore_...   
4  /projects/ps-yeolab5/encore/processing/encore_...   

                                              plus_1  \
0  /projects/ps-yeolab5/encore/processing/encore_...   
1  /projects/ps-yeolab5/encore/processing/encore_...   
2  /projects/ps-yeolab5/encore/processing/encore_...   
3  /projects/ps-yeolab5/encore/processing/encore_...   
4  /projects/ps-yeolab5/encore/processing/encore_...   

                                        plus_control  \
0  /projects/ps-yeolab5/encore/processing/encore_...   
1  /projects/ps-yeolab5/encore/processing/encore_...   
2  /projects/ps-yeolab5/encore/processing/encore_...   
3  /projects/ps-yeolab5/encore/processing/encore_...   
4  /projects/ps-yeolab5/encore/processing/encore_...   

                                             minus_0  \
0  /projects/ps-yeolab5/encore/processing/encore_...   
1  /projects/ps-yeolab5/encore/processing/encore_...   
2  /projects/ps-yeolab5/encore/processing/encore_...   
3  /projects/ps-yeolab5/encore/processing/encore_...   
4  /projects/ps-yeolab5/encore/processing/encore_...   

                                             minus_1  \
0  /projects/ps-yeolab5/encore/processing/encore_...   
1  /projects/ps-yeolab5/encore/processing/encore_...   
2  /projects/ps-yeolab5/encore/processing/encore_...   
3  /projects/ps-yeolab5/encore/processing/encore_...   
4  /projects/ps-yeolab5/encore/processing/encore_...   

                                       minus_control  \
0  /projects/ps-yeolab5/encore/processing/encore_...   
1  /projects/ps-yeolab5/encore/processing/encore_...   
2  /projects/ps-yeolab5/encore/processing/encore_...   
3  /projects/ps-yeolab5/encore/processing/encore_...   
4  /projects/ps-yeolab5/encore/processing/encore_...   

                                               bed_0  \
0  /projects/ps-yeolab5/encore/processing/encore_...   
1  /projects/ps-yeolab5/encore/processing/encore_...   
2  /projects/ps-yeolab5/encore/processing/encore_...   
3  /projects/ps-yeolab5/encore/processing/encore_...   
4  /projects/ps-yeolab5/encore/processing/encore_...   

                                               bed_1  \
0  /projects/ps-yeolab5/encore/processing/encore_...   
1  /projects/ps-yeolab5/encore/processing/encore_...   
2  /projects/ps-yeolab5/encore/processing/encore_...   
3  /projects/ps-yeolab5/encore/processing/encore

In [103]:
encode3.head()


uID      RBP Cell line                                          CLIP_rep1  \
0  203   HNRNPC     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
1  204   RBFOX2     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
2  205  IGF2BP1     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
3  206   HNRNPK     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
4  209    SRSF7     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   

                                           CLIP_rep2  \
0  /projects/ps-yeolab3/encode/analysis/encode_GR...   
1  /projects/ps-yeolab3/encode/analysis/encode_GR...   
2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
3  /projects/ps-yeolab3/encode/analysis/encode_GR...   
4  /projects/ps-yeolab3/encode/analysis/encode_GR...   

                                               INPUT  
0  /projects/ps-yeolab3/encode/analysis/encode_GR...  
1  /projects/ps-yeolab3/encode/analysis/encode_GR...  
2  /projects/ps-yeolab3/encode/analysis/encode_GR...  
3  /projects/ps-yeolab3/encode/analysis/encode_GR...  
4  /projects/ps-yeolab3/encode/analysis/encode_GR...

In [104]:
encode4.loc[encode4['RBP'].isin(essential_gene)]

Unnamed: 0   uid     Batch    RBP                 prefix  \
17          17  4037  batch11a   EEF2  encode4_batch11a.4037   
22          22  4050  batch11b  DDX47  encode4_batch11b.4050   
29          29  4048   batch12  SF3A1   encode4_batch12.4048   
43          43  4107   batch17   EEF2   encode4_batch17.4107   
48          48  4117   batch17  SF3A1   encode4_batch17.4117   

                                                bam_0  \
17  /projects/ps-yeolab5/encore/processing/encore_...   
22  /projects/ps-yeolab5/encore/processing/encore_...   
29  /projects/ps-yeolab5/encore/processing/encore_...   
43  /projects/ps-yeolab5/encore/processing/encore_...   
48  /projects/ps-yeolab5/encore/processing/encore_...   

                                                bam_1  \
17  /projects/ps-yeolab5/encore/processing/encore_...   
22  /projects/ps-yeolab5/encore/processing/encore_...   
29  /projects/ps-yeolab5/encore/processing/encore_...   
43  /projects/ps-yeolab5/encore/processing/encore_...   
48  /projects/ps-yeolab5/encore/processing/encore_...   

                                          bam_control  \
17  /projects/ps-yeolab5/encore/processing/encore_...   
22  /projects/ps-yeolab5/encore/processing/encore_...   
29  /projects/ps-yeolab5/encore/processing/encore_...   
43  /projects/ps-yeolab5/encore/processing/encore_...   
48  /projects/ps-yeolab5/encore/processing/encore_...   

                                               plus_0  \
17  /projects/ps-yeolab5/encore/processing/encore_...   
22  /projects/ps-yeolab5/encore/processing/encore_...   
29  /projects/ps-yeolab5/encore/processing/encore_...   
43  /projects/ps-yeolab5/encore/processing/encore_...   
48  /projects/ps-yeolab5/encore/processing/encore_...   

                                               plus_1  \
17  /projects/ps-yeolab5/encore/processing/encore_...   
22  /projects/ps-yeolab5/encore/processing/encore_...   
29  /projects/ps-yeolab5/encore/processing/encore_...   
43  /projects/ps-yeolab5/encore/processing/encore_...   
48  /projects/ps-yeolab5/encore/processing/encore_...   

                                         plus_control  \
17  /projects/ps-yeolab5/encore/processing/encore_...   
22  /projects/ps-yeolab5/encore/processing/encore_...   
29  /projects/ps-yeolab5/encore/processing/encore_...   
43  /projects/ps-yeolab5/encore/processing/encore_...   
48  /projects/ps-yeolab5/encore/processing/encore_...   

                                              minus_0  \
17  /projects/ps-yeolab5/encore/processing/encore_...   
22  /projects/ps-yeolab5/encore/processing/encore_...   
29  /projects/ps-yeolab5/encore/processing/encore_...   
43  /projects/ps-yeolab5/encore/processing/encore_...   
48  /projects/ps-yeolab5/encore/processing/encore_...   

                                              minus_1  \
17  /projects/ps-yeolab5/encore/processing/encore_...   
22  /projects/ps-yeolab5/encore/processing/encore_...   
29  /projects/ps-yeolab5/encore/processing/encore_...   
43  /projects/ps-yeolab5/encore/processing/encore_...   
48  /projects/ps-yeolab5/encore/processing/encore_...   

                                        minus_control  \
17  /projects/ps-yeolab5/encore/processing/encore_...   
22  /projects/ps-yeolab5/encore/processing/encore_...   
29  /projects/ps-yeolab5/encore/processing/encore_...   
43  /projects/ps-yeolab5/encore/processing/encore_...   
48  /projects/ps-yeolab5/encore/processing/encore_...   

                                                bed_0  \
17  /projects/ps-yeolab5/encore/processing/encore_...   
22  /projects/ps-yeolab5/encore/processing/encore_...   
29  /projects/ps-yeolab5/encore/processing/encore_...   
43  /projects/ps-yeolab5/encore/processing/encore_...   
48  /projects/ps-yeolab5/encore/processing/encore_...   

                                                bed_1  \
17  /projects/ps-yeolab5/encore/processing/encore_...   
22  /projects/ps-yeolab5/encore/processing/encore_...   
29  /projects/ps-y

In [105]:
encode3.loc[encode3['RBP'].isin(essential_gene)]

uID     RBP Cell line                                          CLIP_rep1  \
0    203  HNRNPC     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
15   228   SF3B4     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
21   242   U2AF2      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
23   244   U2AF1      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
26   249  HNRNPU      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
32   272   U2AF2     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
36   281  HNRNPU     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
37   282   U2AF1     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
47   311   XRCC6     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
54   331   RBM22      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
98   480   XRCC6      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
105  494   RBM22     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
110  506   SF3B1      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
111  507   SF3B4      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
125  545   EIF3H     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
127  548  ZRANB2      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
132  556   CDC40     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   
165  646   EIF3G      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
176  677  GEMIN5      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
183  685    WDR3      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
193  699  HNRNPC      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   
215  752   PRPF4     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   

                                             CLIP_rep2  \
0    /projects/ps-yeolab3/encode/analysis/encode_GR...   
15   /projects/ps-yeolab3/encode/analysis/encode_GR...   
21   /projects/ps-yeolab3/encode/analysis/encode_GR...   
23   /projects/ps-yeolab3/encode/analysis/encode_GR...   
26   /projects/ps-yeolab3/encode/analysis/encode_GR...   
32   /projects/ps-yeolab3/encode/analysis/encode_GR...   
36   /projects/ps-yeolab3/encode/analysis/encode_GR...   
37   /projects/ps-yeolab3/encode/analysis/encode_GR...   
47   /projects/ps-yeolab3/encode/analysis/encode_GR...   
54   /projects/ps-yeolab3/encode/analysis/encode_GR...   
98   /projects/ps-yeolab3/encode/analysis/encode_GR...   
105  /projects/ps-yeolab3/encode/analysis/encode_GR...   
110  /projects/ps-yeolab3/encode/analysis/encode_GR...   
111  /projects/ps-yeolab3/encode/analysis/encode_GR...   
125  /projects/ps-yeolab3/encode/analysis/encode_GR...   
127  /projects/ps-yeolab3/encode/analysis/encode_GR...   
132  /projects/ps-yeolab3/encode/analysis/encode_GR...   
165  /projects/ps-yeolab3/encode/analysis/encode_GR...   
176  /projects/ps-yeolab3/encode/analysis/encode_GR...   
183  /projects/ps-yeolab3/encode/analysis/encode_GR...   
193  /projects/ps-yeolab3/encode/analysis/encode_GR...   
215  /projects/ps-yeolab3/encode/analysis/encode_GR...   

                                                 INPUT  
0    /projects/ps-yeolab3/encode/analysis/encode_GR...  
15   /projects/ps-yeolab3/encode/analysis/encode_GR...  
21   /projects/ps-yeolab3/encode/analysis/encode_GR...  
23   /projects/ps-yeolab3/encode/analysis/encode_GR...  
26   /projects/ps-yeolab3/encode/analysis/encode_GR...  
32   /projects/ps-yeolab3/encode/analysis/encode_GR...  
36   /projects/ps-yeolab3/encode/analysis/encode_GR...  
37   /projects/ps-yeolab3/encode/analysis/encode_GR...  
47   /projects/ps-yeolab3/encode/analysis/encode_GR...  
54   /projects/ps-yeolab3/encode/analysis/encode_GR...  
98   /projects/ps-yeolab3/encode/analysis/encode_GR...  
105  /projects/ps-yeolab3/encode/analysis/encode_GR...  
110  /projects/ps-yeolab3/encode/analysis/encode_GR...  
111  /pr

In [106]:
# we found the essential genes

## Fisher Exact Test demo using RBM22

In [107]:
# What are the components of this command?
uid = encode3.loc[(encode3['RBP']=='RBM22')&(encode3['Cell line']=='HepG2'), 'uID'].iloc[0]

In [108]:
uid

'494'

In [109]:
idr_path = '/home/hsher/seqdata/eclip_bed/sorted/{}.01v02.IDR.out.0102merged.bed.blacklist_removed.bed.narrowPeak.bed'.format(uid)

In [110]:
idr_path

'/home/hsher/seqdata/eclip_bed/sorted/494.01v02.IDR.out.0102merged.bed.blacklist_removed.bed.narrowPeak.bed'

In [111]:
idr = BedTool(idr_path)

In [112]:
idr

<BedTool(/home/hsher/seqdata/eclip_bed/sorted/494.01v02.IDR.out.0102merged.bed.blacklist_removed.bed.narrowPeak.bed)>

## Goal:
- does our RBP (RBM22) regulate tumor-essential genes more than other genes? (likelihood)

4 groups we need to find:
1. tumor essential AND bound by RBM22
2. tumor essential and NOT bound by RBM22
3. NOT tumor essential AND bound by RBM22
4. NOT tumor essential and NOT bound by RBM22

Use BedTool intersect to find the 4 groups:

In [113]:
# group 1: essential genes AND bound by RBM22
group_1 = essential_gene_coords.intersect(idr, s = True, u = True).saveas()

In [114]:
# group 2: essential genes and NOT bound by RBM22
group_2 = essential_gene_coords.intersect(idr, s = True, v = True).saveas()

In [115]:
# transcripts that are not tumor essential genes, so we can repeat this process for group 3 & 4
nonessential_gene_coords = transcript_coords.intersect(essential_gene_coords, s = True, v = True).saveas()

In [116]:
# group 3: nonessential genes AND bound by RBM22
group_3 = nonessential_gene_coords.intersect(idr, s = True, u = True).saveas()

In [117]:
# group 4: nonessential genes and NOT bound by RBM22
group_4 = nonessential_gene_coords.intersect(idr, s = True, v = True).saveas()

In [120]:
# counting genes for table
table_for_stat = pd.DataFrame([
    [len(group_1), len(group_2)], 
     [len(group_3), len(group_4)]
], 
    
    columns = ['bound by RBM22', 'not bound'],
    index = ['essential', 'not essential']
)

In [122]:
table_for_stat
# things to note - syntax for putting in rows [square brackets]

bound by RBM22  not bound
essential                  26        102
not essential            1244      58719

Notes from Charlene on stats
- scipy stats documentation: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.fisher_exact.html
- odds ratio from NIH: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2938757/


In [123]:
from scipy.stats import fisher_exact

In [124]:
odds, p = fisher_exact(table_for_stat, alternative = 'two-sided')

In [125]:
odds

12.031823340268584

In [126]:
# how to calculate odds ratio
(26/1244)/(102/58719)

12.031823340268582

In [127]:
p

2.793388556985412e-18

In [128]:
# very low p value means it's significant! :D

Conclusion: we can conclude that tumor essential genes are regulated by RBM22, meaning they have IDR peaks, more than expected